projects/extension/src/containers/Options.tsx

```diff
@@ -67,70 +67,63 @@ export const Options: React.FunctionComponent = () => {
   const [notifications, setNotifications] = useState<boolean>(false)
   const [menu, setMenu] = useState<number>(0)
   const [showModal, setShowModal] = useState<boolean>(false)
-  const [bg, setBg] = useState<Background | undefined>()
   const [actionResult, setActionResult] = useState<string>("")
 
   useEffect(() => {
-    chrome.runtime.getBackgroundPage((backgroundPage) => {
-      setBg(backgroundPage as Background)
-    })
-  }, [])
-
-  useEffect(() => {
-    if (!bg) return
-
     const getNotifications = async () => {
-      const result = await bg?.uiInterface.getChromeStorageLocalSetting(
-        "notifications",
+      setNotifications(
+        (await environment.get({ type: "notifications" })) || false,
       )
-      setNotifications(result?.notifications as SetStateAction<boolean>)
     }
 
     getNotifications()
 
     window.navigator?.brave?.isBrave().then(async (isBrave: any) => {
-      const { braveSetting } =
-        await bg.uiInterface.getChromeStorageLocalSetting("braveSetting")
+      const braveSetting = await environment.get({ type: "braveSetting" })
       setShowModal(isBrave && !braveSetting)
     })
 
     const refresh = () => {
-      const networks = new Map<string, NetworkTabProps>()
-      bg.uiInterface.chains.forEach((chain) => {
-        const { chainName, tab, isSyncing, peers, bestBlockHeight } = chain
-
-        const network = networks.get(chainName)
-        if (!network) {
-          return networks.set(chainName, {
-            name: chainName,
-            health: {
-              isSyncing,
-              peers,
-              status: "connected",
-              bestBlockHeight,
-            },
-            apps: tab ? [{ name: tab.url, url: tab.url }] : [],
-          })
-        }
-
-        if (tab) network.apps.push({ name: tab.url, url: tab.url })
+      environment.get({ type: "activeChains" }).then((chains) => {
+        const networks = new Map<string, NetworkTabProps>()
+        ;(chains || []).forEach((chain) => {
+          const { chainName, tab, isSyncing, peers, bestBlockHeight } = chain
+
+          const network = networks.get(chainName)
+          if (!network) {
+            return networks.set(chainName, {
+              name: chainName,
+              health: {
+                isSyncing,
+                peers,
+                status: "connected",
+                bestBlockHeight,
+              },
+              apps: [{ name: tab.url, url: tab.url }],
+            })
+          }
+
+          if (tab) network.apps.push({ name: tab.url, url: tab.url })
```

Comment:
If tab is always defined then this is not needed:
```suggestion
         network.apps.push({ name: tab.url, url: tab.url })
```